frontend/src/components/assessments/student-experience/Question.tsx

```diff
@@ -9,62 +9,75 @@ import {
   QuestionTextMetadata,
   TextMetadata,
 } from "../../../types/QuestionMetadataTypes";
-import { QuestionElementType } from "../../../types/QuestionTypes";
+import {
+  QuestionElementType,
+  ResponseElementType,
+} from "../../../types/QuestionTypes";
 
-import Image from "./question-components/Image";
-import MultipleChoice from "./question-components/MultipleChoice";
-import MultiSelect from "./question-components/MultiSelect";
-import QuestionText from "./question-components/QuestionText";
-import ShortAnswer from "./question-components/ShortAnswer";
-import Text from "./question-components/Text";
+import Image from "./question-elements/Image";
+import MultipleChoice from "./question-elements/MultipleChoice";
+import MultiSelect from "./question-elements/MultiSelect";
+import QuestionText from "./question-elements/QuestionText";
+import ShortAnswer from "./question-elements/ShortAnswer";
+import Text from "./question-elements/Text";
 
-interface DisplayQuestionProps {
-  questionComponents: QuestionComponentResponse[];
+interface QuestionProps {
+  elements: QuestionComponentResponse[];
 }
 
-const Question = ({
-  questionComponents,
-}: DisplayQuestionProps): React.ReactElement => {
+const Question = ({ elements }: QuestionProps): React.ReactElement => {
+  const answerElementIndex = useMemo(() => {
+    let answersSoFar = 0;
+    return elements.map((element) =>
+      // eslint-disable-next-line no-plusplus
+      element.type in ResponseElementType ? answersSoFar++ : 0,
+    );
```

Comment:
Honestly, given that the linter needs to be disabled for this to work, I think I'd personally prefer a more explicit version:
```suggestion
    return elements.map((element) => {
      if (element.type in ResponseElementType) {
        const elementIndex = answersSoFar;
        answersSoFar += 1;
        return elementIndex;
      }
      return 0;
    });
```

(or change the starting count to `-1` like it was before to save a line... technically, if you use a starting index of `-1`, you can actually do the same `element.type in ResponseElementType ? (answersSoFar += 1) : 0`, but I think that's far too confusing)

It's not bad as is, though; I'm just wary of suppressing our warnings since they are _generally_ good suggestions.

Reply:
sounds good to me! updated (: